### ISSUE TITLE
Enter text in input areas

### ISSUE THREAD

**Issue by jtaylor113** [mozilla/firefox_voice_community#256]
I would like to be able to enter text within text boxes in websites via voice instead of keyboard. Basically, voice to text entry.

**Comment by awallin** [comment]
Great idea @jtaylor113. Can you give me an example of the type of website and forms you want to use it on? What types can you imagine filling out with your voice. And how what phrasing would you use? That info can help us design and implement a feature like this.

**Comment by madb1lly** [comment]
Hi @awallin,

I can imagine some cases where voice dictation for writing an email, postal address or phone number, something like that, might be quicker than entry by keyboard. Password entry could also be useful, although I'm not sure this would be secure unless nobody else is present!

Also having voice control of password autofill from about:logins or Lockwise could also be useful, e.g. up, down, select.

Cheers 🙂

**Comment by awallin** [comment]
Great examples. Thanks @madb1lly.

When you're on a page with a form field you wanted to input with your voice (compose email, input phone#, etc) what would you want to say to start the action?

**Comment by jtaylor113** [comment]
Yes, the dictation feature would work great on some items like address and comment fields.
What about right-clicking on the form field and select "dictate" to start it? As another option, you could add an additional toolbar icon for it - assuming you can tell what form field is focused at the time.

**Comment by madb1lly** [comment]
I suppose I would say "type" followed by the text I want to enter in the field, in the general case.

If I have presets stored, I could say "type address", or "type phone number" etc. The problem is that for addresses there are many different ways the page designer can have laid out the forms. Sometimes its just one box, sometimes its several, sometimes it's not enough!

**Comment by jtaylor113** [comment]
I agree with @madb1lly - saying "type" is a good idea too if you can still maintain focus on the input area when clicking on the microphone or typing control-.
